metrics/src/metrics.rs

```diff
@@ -153,7 +153,8 @@ impl MetricsAgent {
         write_frequency: Duration,
         max_points_per_sec: usize,
     ) -> Self {
-        let (sender, receiver) = unbounded::<MetricsCommand>();
+        const MAX_METRIC_COMMAND_QUEUE_SIZE: usize = 2000;
+        let (sender, receiver) = bounded::<MetricsCommand>(MAX_METRIC_COMMAND_QUEUE_SIZE);
```

Comment:
Before making this channel bounded, we need to find the culprits for too much spam and reduce those if possible.
Otherwise this bounded channel will be filled with spamy metrics at the cost of more important metrics being always dropped.

Reply:
By looking at the log around the metric queue full. We can determine the spammy logs. 

[2022-04-14T17:14:49.036055753Z ERROR solana_metrics::metrics] haha submitting 2561 points
[2022-04-14T17:14:49.040787510Z ERROR solana_metrics::metrics] haha command queue size: 1
[2022-04-14T17:14:49.040845571Z ERROR solana_metrics::metrics] haha command queue size: 2
[2022-04-14T17:14:49.040880957Z ERROR solana_metrics::metrics] haha command queue size: 3
[2022-04-14T17:14:49.040903809Z ERROR solana_metrics::metrics] haha command queue size: 4
[2022-04-14T17:14:49.040934786Z ERROR solana_metrics::metrics] haha command queue size: 5
[2022-04-14T17:14:49.040983423Z ERROR solana_metrics::metrics] haha command queue size: 6
[2022-04-14T17:14:49.041009167Z ERROR solana_metrics::metrics] haha command queue size: 7
[2022-04-14T17:14:49.041039374Z ERROR solana_metrics::metrics] haha command queue size: 8
[2022-04-14T17:14:49.041062290Z ERROR solana_metrics::metrics] haha command queue size: 9
[2022-04-14T17:14:49.041088574Z ERROR solana_metrics::metrics] haha command queue size: 10
[2022-04-14T17:14:49.041111270Z ERROR solana_metrics::metrics] haha command queue size: 11
[2022-04-14T17:14:49.041135302Z ERROR solana_metrics::metrics] haha command queue size: 12
...
[2022-04-14T17:14:49.194224724Z ERROR solana_metrics::metrics] haha command queue size: 1997
[2022-04-14T17:14:49.194255626Z ERROR solana_metrics::metrics] haha command queue size: 1998
[2022-04-14T17:14:49.194304618Z ERROR solana_metrics::metrics] haha command queue size: 1999
[2022-04-14T17:14:49.194336753Z ERROR solana_metrics::metrics] haha command queue size: 2000
[2022-04-14T17:14:49.194395145Z WARN  solana_metrics::metrics] metric queue full, point dropped.
[2022-04-14T17:14:49.194407783Z ERROR solana_metrics::metrics] haha command queue size: 2000
[2022-04-14T17:14:49.194433316Z WARN  solana_metrics::metrics] metric queue full, point dropped.
[2022-04-14T17:14:49.194459347Z ERROR solana_metrics::metrics] haha command queue size: 2000
[2022-04-14T17:14:49.194474065Z WARN  solana_metrics::metrics] metric queue full, point dropped.
[2022-04-14T17:14:49.194487980Z ERROR solana_metrics::metrics] haha command queue size: 2000
[2022-04-14T17:14:49.194509550Z WARN  solana_metrics::metrics] metric queue full, point dropped.
[2022-04-14T17:14:49.194520645Z ERROR solana_metrics::metrics] haha command queue size: 2000
[2022-04-14T17:14:49.194525507Z WARN  solana_metrics::metrics] metric queue full, point dropped.
...
[2022-04-14T17:14:49.496694764Z INFO  solana_metrics::metrics] datapoint: slot_stats_tracking_complete slot=129722522i last_index=0i num_repaired=1i num_recovered=0i min_turbine_fec_set_count=0i is_full=false is_rooted=false is_dead=false
[2022-04-14T17:14:49.496714853Z INFO  solana_metrics::metrics] datapoint: slot_stats_tracking_complete slot=129722521i last_index=0i num_repaired=1i num_recovered=0i min_turbine_fec_set_count=0i is_full=false is_rooted=false is_dead=false
[2022-04-14T17:14:49.496721335Z INFO  solana_metrics::metrics] datapoint: slot_stats_tracking_complete slot=129722520i last_index=0i num_repaired=1i num_recovered=0i min_turbine_fec_set_count=0i is_full=false is_rooted=false is_dead=false
```

Reply:
As I mentioned here: https://github.com/solana-labs/solana/issues/24319#issuecomment-1099544822 
the issue has been reported to be more significant on rpc nodes, but that metric is independent of rpc or not rpc.